test.py

```diff
@@ -1,70 +1,59 @@
+import unittest2 as unittest
 
-import pynwb
-
-flags = {'form': 1, 'pynwb': 2, 'integration': 3}
-
-# setup and parse arguments
-parser = argparse.ArgumentParser('python test.py [options]')
-parser.set_defaults(verbosity=1, suites=[])
-parser.add_argument('-v', '--verbose', const=2, dest='verbosity', action='store_const', help='run in verbose mode')
-parser.add_argument('-q', '--quiet', const=0, dest='verbosity', action='store_const', help='run in verbose mode')
-parser.add_argument('-f', '--form', action='append_const', const=flags['form'], dest='suites',
-                    help='run in verbose mode')
-parser.add_argument('-p', '--pynwb', action='append_const', const=flags['pynwb'], dest='suites',
-                    help='run in verbose mode')
-parser.add_argument('-i', '--integration', action='append_const', const=flags['integration'], dest='suites',
-                    help='run in verbose mode')
-args = parser.parse_args()
-if not args.suites:
-    args.suites = list(flags.values())
-
-# set up logger
-root = logging.getLogger()
-root.setLevel(logging.DEBUG)
-ch = logging.StreamHandler(sys.stdout)
-ch.setLevel(logging.DEBUG)
-formatter = logging.Formatter('======================================================================\n%(asctime)s - %(levelname)s - %(message)s')  # noqa: E501
-ch.setFormatter(formatter)
-root.addHandler(ch)
-
+flags = {'form': 1, 'pynwb': 2, 'integration': 3, 'example': 4}
 
 TOTAL = 0
 FAILURES = 0
 ERRORS = 0
 
-# Run unit tests for form package
-if flags['form'] in args.suites:
-    logging.info('running form unit tests')
-    form_test_result = unittest.TextTestRunner(verbosity=args.verbosity).run(
-        unittest.TestLoader().discover("tests/unit/form_tests"))
-    TOTAL += form_test_result.testsRun
-    FAILURES += len(form_test_result.failures)
-    ERRORS += len(form_test_result.errors)
-
-# Run unit tests for pynwb package
-if flags['pynwb'] in args.suites:
-    logging.info('running pynwb unit tests')
-    pynwb_test_result = unittest.TextTestRunner(verbosity=args.verbosity).run(
-        unittest.TestLoader().discover("tests/unit/pynwb_tests"))
+
+def run_test_suite(directory, description="", verbose=True):
+    global TOTAL, FAILURES, ERRORS
+    logging.info("running %s" % description)
+    pynwb_test_result = unittest.TextTestRunner(verbosity=verbose).run(
+        unittest.TestLoader().discover(directory))
     TOTAL += pynwb_test_result.testsRun
     FAILURES += len(pynwb_test_result.failures)
     ERRORS += len(pynwb_test_result.errors)
 
-# Run integration tests
-if flags['integration'] in args.suites:
-    logging.info('running integration tests')
-    integration_test_result = unittest.TextTestRunner(verbosity=args.verbosity).run(
-        unittest.TestLoader().discover("tests/integration"))
-    TOTAL += integration_test_result.testsRun
-    FAILURES += len(integration_test_result.failures)
-    ERRORS += len(integration_test_result.errors)
 
+def _import_from_file(script):
```

Comment:
I think it is better to:
1) Change the test file names so that they are importable
2) Moving creating-and-writing-nwbfile-2.py  and creating-and-writing-nwbfile.py under tests/integration
3) Importing them when needed rather than "load_source"ing them.
4) Adding some assertions as opposed to just creating files and removing them? It might be better to make sure we check the integrity of the files(without testing the h5py library).

Reply:
Good point. I just created an issue to capture these: https://github.com/NeurodataWithoutBorders/pynwb/issues/270

In the mean time, I suggest we integrate so that we have:
* a "not prefect" but working baseline for improving example testing
* an example documentation that is not broken

Also, since we may move toward ipython notebook for example and tutorials, I would prefer not to spend too much time yet.

Let me know what you think ?

Reply:
Great, I think this is perfectly fine. We can work on #270 later. Otherwise it LGTM.